aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncClient.java

```diff
@@ -525,4 +531,116 @@ public AppSyncStore getStore() {
     public S3ObjectManager getS3ObjectManager() {
         return mS3ObjectManager;
     }
+
+    /**
+     * Provides the ability to sync using a baseQuery, deltaQuery, subscription, and a refresh interval
+     * @param baseQuery the base query to get the baseline state
+     * @param baseQueryCallback callback to handle the baseQuery results
+     * @param subscription subscription to get changes on the fly
+     * @param subscriptionCallback callback to handle the subscription messages
+     * @param deltaQuery the catchup query
+     * @param deltaQueryCallback callback to handle the deltaQuery results
+     * @param baseRefreshIntervalInSeconds time duration (specified in seconds) when the base query will be re-run to get an updated baseline state.
+     * @param <D>
+     * @param <T>
+     * @param <V>
+     * @return a syncId that can be used later to cancel the sync operation.
+     */
+    public <D extends Query.Data, T, V extends Query.Variables> Long sync(
+            @Nonnull Query<D, T, V> baseQuery,
+            GraphQLCall.Callback<Query.Data> baseQueryCallback,
+            Subscription<D,T,V> subscription,
+            AppSyncSubscriptionCall.Callback subscriptionCallback,
+            Query<D,T,V> deltaQuery,
+            GraphQLCall.Callback<Query.Data> deltaQueryCallback,
+            long baseRefreshIntervalInSeconds) {
+        AWSAppSyncDeltaSync helper = new AWSAppSyncDeltaSync(baseQuery,this, applicationContext);
+
+        helper.setBaseQueryCallback(baseQueryCallback);
+
+        helper.setSubscription(subscription);
+        helper.setSubscriptionCallback( subscriptionCallback);
+
+        if ( deltaQuery == null ||deltaQueryCallback == null ) {
+            Log.d(TAG, "One of the following is null - Delta Query or Delta Query callback. Will switch to using the base query & callback");
+            helper.setDeltaQuery(baseQuery);
```

Comment:
If the delta query is not specified and the internet goes off and comes back (but we are still in the sync window) will this now run baseQuery instead?

Reply:
Correct. The synchronization will happen, with the baseQuery being treated as the deltaQuery.